src/crimson/net/ProtocolV2.cc

```diff
@@ -1763,7 +1803,10 @@ void ProtocolV2::trigger_replacing(bool reconnect,
         return write_frame(reconnect_ok);
       } else {
         client_cookie = new_client_cookie;
-        conn.set_peer_name(new_peer_name);
+        assert(conn.get_peer_type() == new_peer_name.type());
+        if (conn.get_peer_id() == entity_name_t::NEW) {
+          conn.set_peer_id(new_peer_name.num());
+        }
```

Comment:
Sorry was on holiday, not able to reply on time..

> `assert (validate_peer_name(new_peer_name));`

It is correct but not appropriate IMO, `validate_peer_name()` is used by the accepting connection to verify if it can replace the existing one. Here the role is reversed.

> `conn.set_peer_id(new_peer_name.num());`

It is not correct because the information of `new_peer_name.num()` may not be available (aka `entity_name_t::NEW`) during handshake.